bundles/org.openhab.core.crypto.cryptostore/src/main/java/org/openhab/core/crypto/cryptostore/CryptoStore.java

```diff
@@ -0,0 +1,328 @@
+        this.cert = new String(Base64.getEncoder().encode(cert.getEncoded()));
+    }
+
+    public Certificate getCert() throws CertificateException {
+        Certificate cert = CertificateFactory.getInstance("X.509")
+                .generateCertificate(new ByteArrayInputStream(Base64.getDecoder().decode(this.cert.getBytes())));
+        return cert;
+    }
+
+    public void setCaCert(String caCert) {
+        this.caCert = caCert;
+    }
+
+    public void setCaCert(Certificate caCert) throws CertificateEncodingException {
+        this.caCert = new String(Base64.getEncoder().encode(caCert.getEncoded()));
+    }
+
+    public Certificate getCaCert() throws CertificateException {
+        Certificate caCert = CertificateFactory.getInstance("X.509")
+                .generateCertificate(new ByteArrayInputStream(Base64.getDecoder().decode(this.caCert.getBytes())));
+        return caCert;
+    }
+
+    public void setAlias(String alias) {
+        this.alias = alias;
+    }
+
+    public String getAlias() {
+        return this.alias;
+    }
+
+    public void setCAAlias(String caAlias) {
+        this.caAlias = caAlias;
+    }
+
+    public String getCAAlias() {
+        return this.caAlias;
+    }
+
+    public void setAlgorithm(String keystoreAlgorithm) {
+        this.keystoreAlgorithm = keystoreAlgorithm;
+    }
+
+    public String getAlgorithm() {
+        return this.keystoreAlgorithm;
+    }
+
+    public void setKeySize(int keySize) {
+        this.keySize = keySize;
+    }
+
+    public int getKeySize() {
+        return this.keySize;
+    }
+
+    public void setKeyLength(int keyLength) {
+        this.keyLength = keyLength;
+    }
+
+    public int getKeyLength() {
+        return this.keyLength;
+    }
+
+    public void setDistName(String distName) {
+        this.distName = distName;
+    }
+
+    public String getDistName() {
+        return this.distName;
+    }
+
+    public void setKeystoreFileName(String keystoreFileName) {
+        this.keystoreFileName = keystoreFileName;
+    }
+
+    public String getKeystoreFileName() {
+        return this.keystoreFileName;
+    }
+
+    public void setKeys(String privKey, byte[] keyString, String cert) throws GeneralSecurityException, Exception {
+        setPrivKey(privKey, keyString);
+        setCert(cert);
+    }
+
+    public void setKeyStore(String keystoreFileName) {
+        this.keystoreFileName = keystoreFileName;
+    }
+
+    public void loadFromKeyStore(String keystoreFileName, String keystorePassword, byte[] keyString)
+            throws GeneralSecurityException, IOException, Exception {
+        this.keystoreFileName = keystoreFileName;
+        loadFromKeyStore(keystorePassword, keyString);
+    }
+
+    public void loadFromKeyStore(String keystorePassword, byte[] keyString)
+            throws GeneralSecurityException, IOException, Exception {
+        Key key = convertByteToKey(keyString);
+        KeyStore keystore = KeyStore.getInstance("JKS");
```

Comment:
Couple of suggestions with regards to keystore type:

1.   Change from JKS format (which can only store asymmetric keys) to "JCEKS" which can store both asym and symmetric keys.  
2. Write the keystore type to a file on the filesystem, perhaps alongside the keystore itself.   This allows for:
  a. migration to a different keystore type in the future if necessary.  (There's some drama in the keystore space - Oracle recently change their default keystore from JKS to PCKS12, and JKS seems to be deprecated at this point)
  b. There may be cases where a FIPS compliant keystore is required which would warrant a different keystore type.  I don't think we need to go all the way down that rabbit hole just yet, but it would be good to have some scaffolding in place

Reply:
What would be the use case for storing symmetric keys?  In general those are generated as needed for things like IPSEC connections.  

I am tracking the Oracle JKS/PKCS12.  We could migrate to PKCS12 if we wanted to.  I had some issues with it early on and JKS just worked.

Reply:
The openhab binding has symmetric (AES) keys for secure communication with the devices.  Looks like PKCS12 has basic support for symmetric (AES) keys.

Reply:
The AES bits there are just for storing the private key in the memory. It's not used at all for interaction with the device.

Reply:
You are referring to the AES key in the CryptoStore.  I am referring to the AES keys in the zwave binding.  If the CryptoStore is going to be adopted, it needs to support storage of symmetric keys.  JKS does not allow that and JKS format is basically deprecated by Oracle.  

Use PCKS12 or JCEKS instead.  Use <keystore_name>.pkcs12 or <keykstore_name>.jceks so the format is obvious.. in case it needs to change in the future.   Best to change the default now instead of just creating a problem to have to solve later